folium/plugins/geocoder.py

```diff
@@ -50,12 +68,24 @@ class Geocoder(JSCSSMixin, MacroElement):
         )
     ]
 
-    def __init__(self, collapsed=False, position="topright", add_marker=True, **kwargs):
+    def __init__(
+        self,
+        collapsed: bool = False,
+        position: str = "topright",
+        add_marker: bool = True,
+        geocode_zoom: int | None = 11,
```

Comment:
> as the default zoom level of 11 is not suitable for close-up locations (and only works when add_marker is False)

Question: is 11 a good default? Or would None be a better default?

And another question: should we check or correct this value when `add_marker` is False?

Reply:
11 would preserve the current behavior so existing users would not be caught by surprise. The real gotcha would be the case when `add_marker`=True (when default handler in leaflet-control-geocoder will be using `fit_bounds` with no regard to current map zoom). What may not be obvious to normal folium users (with no knowledge of JS) is that both the default event handler for `markgeocode` as well as the one defined in `_template` would be triggered.

Not sure if there is a need to correct (beyond keeping the current default of 11). Can you explain more?

Reply:
> Not sure if there is a need to correct (beyond keeping the current default of 11).

That's good then. It was more a question for you. We can keep it like it is now then.